### ISSUE TITLE
Failure in server-side preview on KPT 0.38.0 exits with 0

### ISSUE THREAD

**Issue by fsommar** [GoogleContainerTools/kpt#1470]
Hi 👋 

In the process of upgrading to KPT 0.38.0 in our CI builds, we noticed that doing a server-side preview of invalid resources now has a 0 exit code. Before 0.38.0, the exit code was non-zero. Here's an example setup:

```bash
$ echo >cm.yaml "
apiVersion: v1
kind: ConfigMap
metadata:
  name: ConfigMap
"
$ kpt live init
```

The output using KPT 0.38.0 does say that the resource failed but doesn't provide any additional info, and the exit status is 0.

```bash
$ kpt live preview --server-side .
# configmap/ConfigMap failed (preview-server)
# 1 resource(s) applied. 0 created, 0 unchanged, 0 configured, 1 failed (preview-server)
# 0 resource(s) pruned, 0 skipped, 0 failed (preview-server)

$ echo $?
# 0
```

For comparison, here's the output of KPT 0.37.1, and I would expect the two outputs to be similar-ish (at least the exit code).

```bash
$ kpt0.37.1 live preview --server-side .
# The ConfigMap "ConfigMap" is invalid: metadata.name: Invalid value: "ConfigMap": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')

$ echo $?
# 1
```

**Comment by mikebz** [comment]
thanks for reporting this.  We are going to take a look at it ASAP.

**Comment by mortent** [comment]
@fsommar Thanks for reporting this. I have found the problem and the cause is that kpt used to bail out at the first error, but we now continue even if one resource fails. This is similar to how kubectl works. 
It requires that we handle errors in the printer slightly different and I've started work on a fix. The good news about this, is that when we get a fix in here, preview will actually run preview for all resources even if one fails, instead of just reporting the first resource with a problem.
